lib/crawly/engine.ex

```diff
@@ -10,6 +10,9 @@ defmodule Crawly.Engine do
 
   @type t :: %__MODULE__{started_spiders: started_spiders()}
   @type started_spiders() :: %{optional(module()) => identifier()}
+  @type list_spiders() :: [
```

Comment:
I would suggest to do it slightly differently:

```
@type spider_status() :: :stopped | {:started, identifier()}

# And in the function we could have
@spec list_spiders() :: [%{name: module(), state: spider_status()}]
```

I prefer this type of specs...

Reply:
OK will adjust